pandas/_libs/window.pyx

```diff
@@ -1414,18 +1438,31 @@ def roll_quantile(ndarray[float64_t, cast=True] input, int64_t win,
                     skiplist.insert(val)
 
         if nobs >= minp:
-            idx = int(quantile * <double>(nobs - 1))
-
-            # Single value in skip list
             if nobs == 1:
+                # Single value in skip list
                 output[i] = skiplist.get(0)
-
-            # Interpolated quantile
             else:
-                vlow = skiplist.get(idx)
-                vhigh = skiplist.get(idx + 1)
-                output[i] = ((vlow + (vhigh - vlow) *
-                             (quantile * (nobs - 1) - idx)))
+                idx_with_fraction = quantile * <double> (nobs - 1)
+                idx = int(idx_with_fraction)
```

Comment:
Given everything else here is done with C syntax, can we use that casting convention here?